src/backend/renderer/damage/mod.rs

```diff
@@ -454,17 +461,22 @@ impl OutputDamageTracker {
         output_scale: Scale<f64>,
         output_transform: Transform,
         output_geo: Rectangle<i32, Physical>,
-        damage: &mut Vec<Rectangle<i32, Physical>>,
         render_elements: &mut Vec<&'a E>,
-        opaque_regions: &mut Vec<(usize, Vec<Rectangle<i32, Physical>>)>,
     ) -> RenderElementStates
     where
         E: Element,
     {
+        self.damage.clear();
+        self.element_damage.clear();
+        self.opaque_regions.clear();
+
         let mut element_render_states = RenderElementStates {
             states: HashMap::with_capacity(elements.len()),
         };
 
+        let mut damage = std::mem::take(&mut self.damage);
+        let mut element_damage = std::mem::take(&mut self.element_damage);
```

Comment:
This will cause new Vec's to be created, no? Why are we not directly using `self.damage` here, we just cleared it anyway?

Reply:
It is returned here: https://github.com/Smithay/smithay/blob/13bc4241c52be424fdacf2a60000a45b9e9dae87/src/backend/renderer/damage/mod.rs#L681 and an empty Vec does not allocate. Anyway, this is also just some leftover from searching for a error and we can just use `self.damage`, yeah.

Reply:
Right and empty Vec does nothing, but it is still confusing to read. If there is no reason to do it this way, I would prefer to just use the value directly in all these cases.

Reply:
mostly done, we need to take `element_damage` as we move it around a bit